src/matcha_ml/state/remote_state_manager.py

```diff
@@ -122,6 +126,28 @@ def azure_storage(self) -> AzureStorage:
 
         return self._azure_storage
 
+    def _resource_group_exists(self, resource_group_name: str) -> bool:
```

Comment:
`AzureClient` is being instantiated. `AzureStorage` class also does this. Is the better for this function to stay inside `AzureStorage` or `AzureClient` class?

Reply:
Good idea, unfortunately we cannot use `AzureStorage` to access `AzureClient`, this is because the `AzureStorage` init is as follows:
```
class AzureStorage:
    """Class to interact with Azure blob storage."""

    az_client: AzureClient
    blob_service_client: BlobServiceClient

    def __init__(self, account_name: str, resource_group_name: str) -> None:
        """Initialize Azure Storage.

        Args:
            account_name (str): Azure storage account name
            resource_group_name (str): Name of resource group containing given account name
        """
        self.az_client = AzureClient()
        _conn_str = self.az_client.fetch_connection_string(
            storage_account_name=account_name, resource_group_name=resource_group_name
        )
        self.blob_service_client = BlobServiceClient.from_connection_string(
            conn_str=_conn_str
        )
```

Which runs the following command `self.az_client.fetch_connection_string(storage_account_name=account_name, resource_group_name=resource_group_name)` which throws the error we are trying to prevent. Therefore we need to check for the storage account before the AzureStorage service is initialised.

Reply:
Good point.